api/v3/Case/Getdetailscount.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+/**
+ * Case.getdetailscount API
+ *
+ * Provides a count of cases but properly respects filters unlike `getcount`.
+ * Please refer to C51-277 for more information.
+ *
+ * @param array $params
+ * @return array API result
+ * @throws API_Exception
+ */
+function civicrm_api3_case_getdetailscount($params) {
```

Comment:
If the `getcount` endpoint isn't working as expected then was it not an option to fix that?

Reply:
There were talks between Davi and the team and they agreed that the best solution was to create a new endpoint instead of fixing the core one. I'm not sure about all the details, but some of the reasons were avoiding modifying a core endpoint, and also this endpoint works for multiple Entities. If it's going to be fixed it needs to be a PR from core and I think this was to be avoided.

Reply:
Fair enough, just in general it seems better to fix the original than creating more endpoints to be maintained but if it's been discussed already then it's fine.